### PR TITLE
Require webpack_config field to build with custom configuration

### LINKED ISSUE #296: Require webpack_config field in wrangler.toml to use custom webpack configuration
It is not abundantly clear that wrangler will use your custom `webpack.config.js` to package everything. Some users may want to start using wrangler with a project that already has a `webpack.config.js`, but they may not understand that wrangler will use that configuration over its own default. We should require users who want to use their own configuration to specify it in their `wrangler.toml` `webpack_config` field so that there is no confusion that their webpack configuration is impacting their build output.

We need to decide if this is something we want to do, or if we want to leave things like they are and automatically override wrangler’s default webpack behavior when a user has their own `webpack.config.js`

**Edit**

Implementing this would be a breaking change for users that currently use `webpack.config.js` to build their workers, they would need to add a `webpack_config` field to their `wrangler.toml` in order for it to continue working. We would want to add an error message if they have a file `webpack.config.js` to say that if they want it to be used, they should add it to their configuration to override the default.

We're looking at implementing this for the upcoming release because some users will use `webpack.config.js` to build other types of apps alongside their worker. In these cases, the user will want to keep `webpack.config.js` around, but not to build their worker

### PR DESCRIPTION
**Pull request by EverlastingBugstopper**
Fixes #296 and #833 

As promised - we are deprecating inferred usage of `webpack.config.js` in 1.6.0 and are requiring the `webpack_config` file to be set in order to use a custom webpack configuration file. 

```console
$ wrangler build
⚠️  If you would like to use your own custom webpack configuration, you will need to add this to your wrangler.toml:
webpack_config = "webpack.config.js"
✨  Built successfully, built project size is 535 bytes.
```

```console
$ wrangler build
using custom
✨  Built successfully, built project size is 535 bytes.
```

```console
$ wrangler build --env test
⚠️  If you would like to use your own custom webpack configuration, you will need to add this to your wrangler.toml:
webpack_config = "webpack.config.js"
✨  Built successfully, built project size is 535 bytes.
```

```console
$ wrangler build --env test
using custom config
✨  Built successfully, built project size is 535 bytes.
```
      
```console
$ wrangler build
⚠️  Workers Sites does not support custom webpack configuration files
✨  Built successfully, built project size is 11 KiB.
```

### CODE REVIEW

File: tests/build.rs [hunk 1]
```diff
@@ -139,21 +143,8 @@ fn it_fails_with_multiple_webpack_configs() {
...
     single_env_settings! {fixture, r#"
-        type = "Webpack"
-    "#};
-
-    build_fails_with(fixture, "Multiple webpack configurations are not supported. You can specify a different path for your webpack configuration file in wrangler.toml with the `webpack_config` field");
-    utils::cleanup(fixture);
-}
-
-#[test]
-fn it_fails_with_multiple_specify_webpack_configs() {
```
Comment (ashleymichal) [inline tests/build.rs]: should this test still exist? it's kind of redundant but i don't think its outcome is changed by this PR.

Comment (EverlastingBugstopper) [reply tests/build.rs]: It's the same test as `it_fails_with_multiple_webpack_configs` the only difference previously was one had inferred `webpack.config.js` existed and one had it in the toml - after this breaking change they are all "specify" tests

**Comment by EverlastingBugstopper** [comment]
> the utils removal is a step backwards, kind of, but makes the status quo consistent so is fine.

I brought back the util and moved a couple of other tests to it - the reason I removed it earlier was because I didn't realize that I could use the webpack config file created programatically in the wrangler.toml

**Comment by motiejunas** [comment]
Could you give more details on "Workers Sites does not support custom webpack configuration files"? Basically it means that I cannot use TypeScript for Workers Sites anymore?

**Comment by ashleymichal** [comment]
@motiejunas the error message is historical; when wrangler implicitly looked at the root for a custom webpack config, it made sense to disable the functionality for sites since the root would likely contain a config for the app you were deploying, not for the workers site. with this change, though, we should allow custom webpack configs since you are required to declare them in your wrangler.toml. we'll track it in a separate issue.

**Pull request merged**